src/JuMPContainer.jl

```diff
@@ -17,19 +16,36 @@ end
 #= Generated on the fly
 type JuMPArray{T}
     innerArray::Array{T,N}
-    name::String
-    indexsets
+    meta::Dict{Symbol,Any}
 end
 =#
 type JuMPDict{T,N} <: JuMPContainer{T,N}
     tupledict::Dict{NTuple{N,Any},T}
-    name::Symbol
+    meta::Dict{Symbol,Any}
+
+    JuMPDict() = new(Dict{NTuple{N,Any},T}(), Dict{Symbol,Any}())
+end
+
+function JuMPDict{T,N}(d::Dict{NTuple{N,Any},T})
+    tmp = JuMPDict{T,N}()
+    tmp.tupledict = d
+    tmp
+end
+
+function JuMPDict{T,N}(d::Dict{NTuple{N,Any},T}, meta::Dict{Symbol,Any})
+    tmp = JuMPDict{T,N}()
+    tmp.tupledict = d
+    tmp.meta = meta
```

Comment:
Not ideal that we have to construct two unnecessary (empty) Dicts here, but I couldn't figure out how to make the inner constructors work like I wanted them to.